### DISCUSSION TITLE
LwM2M plugin in combination with auth_psql plugin

### DISCUSSION THREAD

**Discussion by pakokol** [emqx/emqx#7223] [Q&A] [answered]
Hi everyone!

I have problems with configuring emqx_lwm2m and emqx_auth_psql plugins together to achieve a username/password authentication. I was curious if that is even possible?

BR,
Patrik Kokol

**Comment by zmstone** [comment, marked as answer]
@HJianBo @savonarola could you help to have a look.

**Reply by hjianbo** [reply]
In emqx 4.x version this is not possible, only `emqx-auth-http` can be used to do this.

Since LwM2M is not designed with authentication-related fields (e.g. Password, Token, etc.), we can only send the LwM2M Client's  Endpoint Name to a web server to determine if it has login privileges.

i.e, In emqx_auth_http.conf:
```
auth.http.auth_req.params = endpoint_name=%c
```
